README.md

```diff
@@ -2,22 +2,97 @@
 
-## Usage
+This action creates an environment capable of building PROS projects and templates, and builds them using [build.sh](/build-tools/build.sh)
 
-Here are the basics `steps` you'd need to build a pros project:
+Instructions on creating a custom build script, adding additional packages, and using this image as a base are located at the end of this readme.
+
+## Usage:
+
+### Inputs
+
+- `multithreading`
+  - Wether to use multithreading when building the project
+  - Default: `true`
+  - Required: `false`
+- `no_commit_hash`
+  - Wether to include a shortened commit hash at the end of the artifact name
+  - Example: `LemLib@0.5.1+5881ac`
+  - Default: `true`
+  - Required: `false`
+- `copy_readme_and_license_to_include`
+  - Whether to make a VERSION file, copy the README(.md), and copy the LICENSE(.md) files to the `/include/(library name)` folder.
+  - required: `false`
+  - default: `false`
+- `lib_folder_name`
+  - The name of the library's folder name under the include directory.
+  - required: `if copy_readme_and_license_to_include is set`
+- `write_job_summary`
+  - Whether to output build information to GitHub's Job Summary
+  - required: `false`
+  - default: `true`
+
+### Outputs
+
+- `name`
+  - The name of the uploaded artifact.
+
+### Example Workflow
 
 ```yml
-steps:
-  - uses: actions/checkout@v4
-  - uses: LemLib/pros-build@v1
+name: PROS Build Example
+
+on:
+  push:
+    branches: "**"
+  pull_request:
+    branches: "**"
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Run LemLib/pros-build
+        id: test
+        uses: LemLib/pros-build
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.test.outputs.name }}
+          path: ${{ github.workspace }}/template/*
 ```
 
-If your pros project contains a library, this action can automatically build it as a template, and then upload it as an artifact.
+## Using the Container in your own build script
 
-```yml
-steps:
-  - uses: actions/checkout@v4
-  - uses: LemLib/pros-build@v1.0.0
-    with:
-      library-path: lemlib # make sure to substitute this with the correct path
+If you wish to use your own build script using this container as a base, you can do so with the following:
+
+It by default includes the packages built into the Ubuntu docker image, and contains the additional packages below:
+
+```
+jq (Used to parse the GitHub API response in build.sh)
+wget (Used to download the toolchain)
+git (Used to get the HEAD SHA hash)
+gawk (Used to get lines from the user project's Makefile)
+python3-minimal (Minimal installation of Python used for pros-cli)
+python3-pip (Used to install pros-cli in the Dockerfile)
+unzip (Unzips the template so that it can be uploaded to Github Actions)
+pros-cli (through python)
+```
+
+### Editing the Dockerfile
+```Dockerfile
+FROM ghcr.io/LemLib/pros-build:main
```

Comment:
we should probably have a stable branch instead, or maybe can this refer to the latest tag?

Reply:
yeah once we merge it in we should make a stable branch, and update the test action to change both branches

Reply:
we could have it refer to the latest tags if we wanted to.

Reply:
Updated to say :stable instead of :main in next commit